### PR TITLE
Fix domain_access action add/remove logic

### PR DESCRIPTION
**Pull request by zerolab**
`$this->configuration['domain_id']` is a single value, not an array.

So bulk adding or removing content to a domain fails.

### CODE REVIEW

**Comment by agentrickard** [comment]
Any idea why the tests failed?

**Comment by zerolab** [comment]
Partly because of #219 (now rebased).

The one other failure is rather interesting:

```
Status    Group      Filename          Line Function
--------------------------------------------------------------------------------
Pass      Other      DomainTestBase.ph   39 Drupal\domain\Tests\DomainTestBase-
    Enabled modules: domain, domain_access, field, node
Pass      Other      DomainTestBase.ph   43 Drupal\domain\Tests\DomainTestBase-
    Created content type article.
Pass      Other      DomainAccessPermi   38 Drupal\domain_access\Tests\DomainAc
    Created content type page.
Pass      Other      DomainTestBase.ph  131 Drupal\domain\Tests\DomainTestBase-
    Created 5 new domains.
Pass      Role       DomainAccessPermi  154 Drupal\domain_access\Tests\DomainAc
    Created role ID &#039;frjish9g&#039; with name &#039;qqfg&gt;&amp;yN&#039;.
Pass      Role       DomainAccessPermi  154 Drupal\domain_access\Tests\DomainAc
    Created permissions: access content, create page content on assigned
    domains
Pass      User login DomainAccessPermi  154 Drupal\domain_access\Tests\DomainAc
    User created with name jDmjxCED and pass WGqCi6tV2W
Pass      Browser    DomainTestBase.ph  191 Drupal\domain\Tests\DomainTestBase-
    GET http://four.localhost:8080/user/login returned 200 (6.34 KB).
Pass      Browser    DomainTestBase.ph  191 Drupal\domain\Tests\DomainTestBase-
    Valid HTML found on "http://four.localhost:8080/user/login"
Pass      Browser    DomainTestBase.ph  191 Drupal\domain\Tests\DomainTestBase-
    GET http://localhost:8080/user/2 returned 200 (5.42 KB).
Pass      Browser    DomainTestBase.ph  191 Drupal\domain\Tests\DomainTestBase-
    Valid HTML found on "http://localhost:8080/user/2"
Pass      User login DomainTestBase.ph  197 Drupal\domain\Tests\DomainTestBase-
    User jDmjxCED successfully logged in.
Pass      Browser    DomainAccessPermi  169 Drupal\domain_access\Tests\DomainAc
    GET http://four.localhost:8080/node/add/page returned 403 (4.53 KB).
Pass      Browser    DomainAccessPermi  169 Drupal\domain_access\Tests\DomainAc
    Valid HTML found on "http://four.localhost:8080/node/add/page"
Fail      Browser    DomainAccessPermi  171 Drupal\domain_access\Tests\DomainAc
    HTTP response expected 200, actual 403
```

The part to pay attention to:

```
Pass      Browser    DomainTestBase.ph  191 Drupal\domain\Tests\DomainTestBase-
    Valid HTML found on "http://four.localhost:8080/user/login"
Pass      Browser    DomainTestBase.ph  191 Drupal\domain\Tests\DomainTestBase-
    GET http://localhost:8080/user/2 returned 200 (5.42 KB).
```

`domainLogin()` POSTs to four.localhost:8080/user/login, but then gets redirected to http://localhost:8080/user/2. i.e. got logged in to the main domain, thus the 403.

**Comment by agentrickard** [comment]
This might be a behavior change in the Drupal version? That domainLogin() method is a bit of a hack around core testing limitations.

This does appear to be green now, though.

**Comment by zerolab** [comment]
Pushed a couple more commits. The last one goes on #215's toes to remove the test results piping to `tee` which gives false positives.

Tested locally with 8.1.x and getting only one fail. I see there are more fails on Travis.

/me scratching head

**Comment by agentrickard** [comment]
I also got one fail locally (the domainLogin() issue).

I'm going to merge the great work on linting and then we can proceed.

File: domain/src/Tests/DomainTestBase.php [hunk 1]
```diff
@@ -138,7 +138,7 @@ public function domainCreateTestDomains($count = 1, $base_hostname = NULL, $list
    *
    * @return bool
    */
-  protected function drupalUserIsLoggedIn($account) {
+  protected function drupalUserIsLoggedIn(UserInterface $account) {
```
Comment (agentrickard) [inline domain/src/Tests/DomainTestBase.php:141]: This violates the base class and causes tests to fail.

```
Strict Standards: Declaration of Drupal\domain\Tests\DomainTestBase::drupalUserIsLoggedIn() should be compatible with Drupal\simpletest\WebTestBase::drupalUserIsLoggedIn($account) in /home/travis/build/agentrickard/domain/domain/src/Tests/DomainTestBase.php on line 15
```

**Comment by agentrickard** [comment]
I cleaned this up and moved it to #232

**Pull request closed without merging**